plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -74,26 +75,30 @@ export class PersonState {
         this.statsd = statsd
         this.personManager = personManager
 
-        // Used to avoid unneeded person fetches.
-        // :KLUDGE: May change through `handleIdentifyOrAlias` flow on merges/create person
-        this.person = person
+        // Used to avoid unneeded person fetches and to respond with updated person details
+        // :KLUDGE: May change through these flows.
+        this.personContainer = personContainer
 
         // If set to true, we'll update `is_identified` at the end of `updateProperties`
         // :KLUDGE: This is an indirect communication channel between `handleIdentifyOrAlias` and `updateProperties`
         this.updateIsIdentified = false
     }
 
-    async update(): Promise<Person | undefined> {
+    async update(): Promise<LazyPersonContainer> {
```

Comment:
When would we ever not want the previous personContainer not to get updated? Why not just make this `PersonState` class have the logic around lazy fetching etc. This being separate feels more complex than it needs to be & some assumptions are tricky.

Reply:
Not sure what your proposal here is, can you put it into code?

Reply:
As discussed, removed .set() and made person-state truly functional :)